### PR TITLE
[Issue #37]: finish e2e tests

### LINKED ISSUE #37: [Task]: Finish e2e tests
### Summary

They were started here: https://github.com/HHS/simpler-grants-gov/issues/1757

But we need to finish the rest of them to match as many of the bug bash cases as we can https://docs.google.com/spreadsheets/d/1EoBeuiGopBFoI5DRDi3NJ2-Y8lIKMde4RsAaIUmzLC8/edit#gid=0

### Acceptance criteria

- Full set of e2e tests that match bug bash tests

### PR DESCRIPTION
**Pull request by rylew1**
## Summary
Fixes #37 
## Changes proposed
- add some of the relevant tests from bug bash

to run just search tests locally in ui:

`npm run test:e2e:ui -- tests/e2e/search/search.spec.ts`

https://github.com/navapbc/simpler-grants-gov/assets/93001277/d9ffa877-bee5-4320-9eea-6621fda68477

### CODE REVIEW

**Review by acouch — APPROVED** [review]
@rylew1 tested locally, looks great.

(suggestion): The e23 tests now require the API to work, so we could update the docs in a separate ticket or just knock it out here: https://github.com/navapbc/simpler-grants-gov/blob/main/frontend/README.md#-end-to-end-e2e-testing

**Comment by rylew1** [comment]
@acouch - updated docs - will need a reapproval

File: frontend/README.md [hunk 1]
```diff
@@ -144,7 +144,9 @@ To run E2E tests using VS Code:
 1. Download the VS Code extension described in these [Playwright docs](https://playwright.dev/docs/running-tests#run-tests-in-vs-code)
 2. Follow the [instructions](https://playwright.dev/docs/getting-started-vscode#running-tests) Playwright provides
 
-In CI, the "Front-end Checks" workflow (`.github/workflows/ci-frontend.yml`) summary will include an "Artifacts" section where there is an attached "playwright-report". [Playwright docs](https://playwright.dev/docs/ci-intro#html-report) describe how to view HTML Report in more detail.
+Playwright E2E tests run "local-to-local", requiring both the frontend and the API to be running for the tests to pass - and for the database to be seeded with data.
```
Comment (rylew1) [inline frontend/README.md:147]: this is the only new change after approval

Comment (acouch) [reply frontend/README.md:147]: I was thinking of adding to the instructions:

```diff
+ - `cd ../api && make init db-seed-local start` (prerequisite to start the API)
  - `npx playwright install --with-deps`
```

**Review by acouch — APPROVED** [review]

File: frontend/README.md (hunk 1 above)
Comment (rylew1) [reply frontend/README.md:147]: Oh I saw the approval and merged very quickly - can open a small follow on

**Pull request merged**